__tests__/setup/web/mockWebSpeech.js

```diff
@@ -0,0 +1,297 @@
+const { defineEventAttribute } = (EventTarget = window.EventTargetShim);
+const NULL_FN = () => 0;
+
+function createSpeechRecognitionResults(isFinal, transcript) {
+  const results = [
+    [
+      {
+        confidence: 0.9,
+        transcript
+      }
+    ]
+  ];
+
+  results[0].isFinal = isFinal;
+
+  return results;
+}
+
+function createProducerConsumer() {
+  const consumers = [];
+  const queue = [];
+
+  return {
+    cancel() {
+      queue = [];
+    },
+    consume(consumer) {
+      consumers.push(consumer);
+      queue.length && consumers.shift()(...queue.shift());
+    },
+    hasConsumer() {
+      return !!consumers.length;
+    },
+    peek() {
+      return queue[0];
+    },
+    produce(...args) {
+      queue.push(args);
+      consumers.length && consumers.shift()(...queue.shift());
+    }
+  };
+}
+
+const speechRecognitionQueue = createProducerConsumer();
+const speechSynthesisQueue = createProducerConsumer();
+
+class SpeechRecognition extends EventTarget {
+  constructor() {
+    super();
+
+    this.grammars = null;
+    this.lang = 'en-US';
```

Comment:
Just verifying understanding: This assumption on locale is only in regards to our tests, right?

Reply:
Yap. We should add test to make sure Web Chat set `lang` correctly.

Reply:
Added to our list.